include/swift/ABI/Metadata.h

```diff
@@ -540,7 +580,7 @@ struct TargetMetadata {
 
 #if SWIFT_OBJC_INTEROP
```

Comment:
This is great! As a followup, can we eliminate some or all of the `#if SWIFT_OBJC_INTEROP` conditional compilations in these headers? (It would make sense for runtime implementation to still use `#if SWIFT_OBJC_INTEROP`, to implement operations that fundamentally rely on the ObjC runtime, but it'd be cool to make as much functionality as possible dependent only on the template.)

Reply:
One thing that wasn't clear to me is how much of the code depends on definitions in objc system headers, but I'm happy to eliminate as much of the conditionals as possible.